test/Azure.Functions.Cli.Tests/E2E/Helpers/CliTester.cs

```diff
@@ -103,36 +113,48 @@ private static async Task InternalRun(string workingDir, RunConfiguration[] runC
 
                     if (runConfiguration.ExpectExit || (i + 1) < runConfiguration.Commands.Length)
                     {
-                        var exitCode = await exe.RunAsync(logStd, logErr, timeout: runConfiguration.CommandTimeout);
-                        exitError |= exitCode != 0;
+                        exitCode = await exe.RunAsync(logStd, logErr, timeout: runConfiguration.CommandTimeout);
```

Comment:
I can give it a shot. It seems like it's a delay to let some commands execute rather than waiting for the host to be "running"? I also don't know what this 'startHost' logic exists so I may look into how tests are using it and replace it with my new "WaitForHostRunning" flag in run config